### PR TITLE
Add clinical analytics app

### PR DESCRIPTION
**Pull request by ycaokris**
Issue for app: https://github.com/plotly/dash-sample-apps/issues/54

# App pull request
- [x] This is a new app
- [ ] I am improving an existing app (redesigns/code "makeovers")

## About

- Playground deployment URL: https://dash-playground.plotly.host/dash-clinical-analytics/
~~- Current gallery app URL: (delete this line if inapplicable)~~

## Workflow

- [x] I have created a branch in the appropriate monorepo, and the
  elements necessary for successful deployment are in place.
- [x] If the app is a redesigned and/or restyled version of an
  existing gallery app, I've summarized the changes requested in the
  appropriate Streambed issue and confirm that they have been applied.
- [ ] If the app is on the Dash Gallery portal, I have added a link to
  the GitHub repository for the source code in the portal description.
~~- [ ] If the app is a reimplementation of a Python gallery app for the
  DashR gallery, the app in this PR mimics, as closely as possible,
  the style and functionality of the existing app.~~

## The pre-review review

I have addressed all of the following questions:

- [x] Does everything in my code serve some purpose? (I have removed
  any dead and/or irrelevant code.)
- [x] Does everything in my code have a clear purpose? (My code is
  readable and, where it isn't, it has been commented appropriately.)]
- [x] Am I reinventing the wheel? (I have used appropriate packages to
  lessen the volume of code that needs to be maintained.)

### CODE REVIEW

File: apps/dash-clinical-analytics/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+```
+venv\Scripts\activate
+```
+
+Install all required packages by running:
+```
+pip install -r requirements.txt
+```
+
+Run this app locally by:
```
Comment (shammamah) [inline apps/dash-clinical-analytics/README.md]:
```suggestion
Run this app locally with:
```

File: apps/dash-clinical-analytics/app.py [hunk 2]
```diff
@@ -0,0 +1,688 @@
...
+admit_list = df["Admit Source"].unique().tolist()
+
+# Date
+# Format checkin Time
+df["Check In Time"] = df["Check In Time"].apply(
+    lambda x: dt.strptime(x, "%Y-%m-%d %I:%M:%S %p")
+)  # String -> Datetime
+
+# Insert weekday and hour of checkin time
+df["Days of Wk"] = df["Check In Hour"] = df["Check In Time"]
```
Comment (shammamah) [inline apps/dash-clinical-analytics/app.py]:
```suggestion
df["Days of Wk"] = df["Check-In Hour"] = df["Check-In Time"]
```

File: apps/dash-clinical-analytics/app.py [hunk 3]
```diff
@@ -0,0 +1,688 @@
...
+df = pd.read_csv(DATA_PATH.joinpath("clinical_analytics.csv"))
+
+clinic_list = df["Clinic Name"].unique()
+
+df["Admit Source"] = df["Admit Source"].fillna("Not Identified")
+admit_list = df["Admit Source"].unique().tolist()
+
+# Date
+# Format checkin Time
+df["Check In Time"] = df["Check In Time"].apply(
```
Comment (shammamah) [inline apps/dash-clinical-analytics/app.py]:
```suggestion
df["Check-In Time"] = df["Check-In Time"].apply(
```

File: apps/dash-clinical-analytics/app.py [hunk 4]
```diff
@@ -0,0 +1,688 @@
...
+    "Monday",
+    "Tuesday",
+    "Wednesday",
+    "Thursday",
+    "Friday",
+    "Saturday",
+    "Sunday",
+]
+
+check_in_duration = df["Check In Time"].describe()
```
Comment (shammamah) [inline apps/dash-clinical-analytics/app.py]:
```suggestion
check_in_duration = df["Check-In Time"].describe()
```

Didn't know about this `describe()` function -- super cool!

File: apps/dash-clinical-analytics/app.py [hunk 5]
```diff
@@ -0,0 +1,688 @@
...
+
+# Register all departments for callbacks
+all_departments = df["Department"].unique().tolist()  # 17
```
Comment (shammamah) [inline apps/dash-clinical-analytics/app.py]: Is 17 just the number of departments there should be?

File: apps/dash-clinical-analytics/app.py [hunk 6]
```diff
@@ -0,0 +1,688 @@
...
+]
+score_inputs = [Input((i + "_score_graph"), "selectedData") for i in all_departments]
+
+
+def description_card():
+    return html.Div(
+        id="description-card",
+        children=[
+            html.H5("Clinical Analytics"),
+            html.H3("Welcome to Clinical Analytics Dashboard"),
```
Comment (shammamah) [inline apps/dash-clinical-analytics/app.py]:
```suggestion
            html.H3("Welcome to the Clinical Analytics Dashboard"),
```

File: apps/dash-clinical-analytics/app.py [hunk 7]
```diff
@@ -0,0 +1,688 @@
...
+            html.Div(
+                id="intro",
+                children="Explore clinic patient volume by time of the day, waiting time, and care score. Click on heatmap to visualize patient experience at different time points.",
```
Comment (shammamah) [inline apps/dash-clinical-analytics/app.py]:
```suggestion
                children="Explore clinic patient volume by time of day, waiting time, and care score. Click on the heatmap to visualize patient experience at different time points.",
```

File: apps/dash-clinical-analytics/app.py [hunk 8]
```diff
@@ -0,0 +1,688 @@
...
+        id="control-card",
+        children=[
+            html.P("Select Clinic"),
+            dcc.Dropdown(
+                id="clinic-select",
+                options=[{"label": i, "value": i} for i in clinic_list],
+                value=clinic_list[0],
+            ),
+            html.Br(),
+            html.P("Select Checkin Time"),
```
Comment (shammamah) [inline apps/dash-clinical-analytics/app.py]:
```suggestion
            html.P("Select Check-In Time"),
```

File: apps/dash-clinical-analytics/app.py [hunk 9]
```diff
@@ -0,0 +1,688 @@
...
+            html.H3("Welcome to Clinical Analytics Dashboard"),
+            html.Div(
+                id="intro",
+                children="Explore clinic patient volume by time of the day, waiting time, and care score. Click on heatmap to visualize patient experience at different time points.",
+            ),
+        ],
+    )
+
+
+def generate_control_card():
```
Comment (shammamah) [inline apps/dash-clinical-analytics/app.py]: Please add docstrings for all other functions too!

File: apps/dash-clinical-analytics/app.py [hunk 10]
```diff
@@ -0,0 +1,688 @@
...
+    hour_of_day = ""
+    weekday = ""
+    shapes = []
+
+    if hm_click is not None:
+        hour_of_day = hm_click["points"][0]["x"]
+        weekday = hm_click["points"][0]["y"]
+
+        # Add shapes
+        x0 = (x_axis.index(hour_of_day)) * 1 / 24
```
Comment (shammamah) [inline apps/dash-clinical-analytics/app.py]: Why not just 
```suggestion
        x0 = (x_axis.index(hour_of_day)) / 24
```

File: apps/dash-clinical-analytics/app.py [hunk 11]
```diff
@@ -0,0 +1,688 @@
...
+        x1 = x0 + 1 / 24
+        y0 = (y_axis.index(weekday)) * 1 / 7
```
Comment (shammamah) [inline apps/dash-clinical-analytics/app.py]:
```suggestion
        y0 = (y_axis.index(weekday)) / 7
```

File: apps/dash-clinical-analytics/app.py [hunk 12]
```diff
@@ -0,0 +1,688 @@
...
+
+
+def create_table_figure(
+    department, filtered_df, category, category_xrange, selected_index
+):
+    aggregation = {
+        "Wait Time Min": "mean",
+        "Care Score": "mean",
+        "Days of Wk": "first",
+        "Check In Time": "first",
```
Comment (shammamah) [inline apps/dash-clinical-analytics/app.py]:
```suggestion
        "Check-In Time": "first",
```

File: apps/dash-clinical-analytics/app.py [hunk 13]
```diff
@@ -0,0 +1,688 @@
...
+        "Check In Hour": "first",
```
Comment (shammamah) [inline apps/dash-clinical-analytics/app.py]:
```suggestion
        "Check-In Hour": "first",
```

**Comment by shammamah** [comment]
Something weird is happening with the dot plot: 

2019-06-17 16 28 56 dash-playground plotly host 2bb16a37797c

File: apps/dash-clinical-analytics/app.py (hunk 5 above)
Comment (ycaokris) [reply apps/dash-clinical-analytics/app.py]: Yes it's total number of department at initial call. This comment could be removed.

**Comment by ycaokris** [comment]
> Something weird is happening with the dot plot:
> 
> 2019-06-17 16 28 56 dash-playground plotly host 2bb16a37797c

The mis-alignment should be fixed. Figures in table wasn't resized/ expand to parent div properly when callback get fired inside columns. More has been discussed in this community thread https://community.plot.ly/t/how-to-force-dcc-graph-to-resize/14852/6 and I adjusted by a clientside hack.

**Pull request merged**